enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,642 @@
+  # Base, human-readable test name. 
+  "name": "openssl version compliance",
+
+  # Optional subcomponent information.
+  "subcomponent": "ui",
+    
+  # Labels are text strings will can be used like 
+  # Ginkgo labels to group and run tests. Until labels
+  # are part of bigquery data, labels will be appended
+  # to test names with "[label]" so that name based
+  # suite construction will work. 
+  "labels": [
+    "sig-..."
+  ],
+  
+  # Tags are key=value pairs that can be used to 
+  # further classify tests.
+  "tags": {
+    "key": "value"
+  },
+  
+  "resources": {
+    # Tests can request to be run in isolation
+    # from others at different levels:
+    # "instance" - avoids being called along with a conflicting
+    #              test in a single run-test invocation.
+    # "exec"     - avoids being called at the same time in a
+    #              across all parallel extension binary invocations.
+    # "bucket"   - avoids being called in the same planning bucket.
+    # If two tests have a conflict name in common, the isolation
+    # mode will apply. "*" is a special conflict name will 
+    # ensures complete isolation in the requested mode (e.g.
+    # guaranteeing the test will be the only one to
+    # a given invocation of run-test if mode=instance).
+    "isolation": {
+      "mode": "exec",
+      "conflict":  [
+        "gpu",
+      ]
+    },
+      
+    # Tests that require significant resources can identify
+    # that fact so that the planning algorithm will not
+    # overdraw on allocatable pod resources for a set of
+    # tests run in parallel.
+    "memory": "1Gi",
+    
+    # If a duration can be roughly predicted, inclusion of this
+    # information may improve the test execution 
+    # planning algorithm performance.  
+    "duration": "2s",
+      
+    # Timeout information may also support efficient
+    # planning.
+    "timeout": "16s",
+  },
+
+  # Tests can be identified as "informing" or "production".
+  # Informing tests will not negatively impact default views
+  # in Component Readiness.
+  # However, policies may be established in the future that 
+  # cause tests that remain informing too long to be treated
+  # as if they are production.
+  "lifecycle": "informing",
+  
+}
+```
+
+##### Run-Test - Running Extension Tests
+The `run-test` verb is used to cause the extension binary to actually run 
+discovered tests. `run-test` accepts test environment arguments as discussed
+in the "list" verb section (in order to reduce fact discovery time needed
+for each test run) and one or more test names to invoke.
+```
+$ extension-binary run-test 
+  --platform      The hardware or cloud platform ("aws", "gcp", "metal", ...).
+  ...other context arguments...
+  --config        The component configuration profile to assume is active on the cluster.
+  --name | -n     Test name to invoke (-n can be specified multiple times).
+```
+
+`run-test` should run the specified tests in parallel. To prevent 

```

Comment:
This is interesting.  We'll have to rework the parallelism logic on origin to support this, but should reduce the overhead of having to call these binaries individually for each test.